### PR TITLE
fix to add-to-cart handler

### PR DESCRIPTION
**Pull request by newsiberian**
Hello, @mikemurray, I remember what you said about not making changes in `productDetails`, but I'm sorry, I could not stand :smile_cat:

By the way, could you explain what is this snippet for?:
```javascript
if (result && count === 0) {
  address = Session.get("address");
  if (!address) {
    return locateUser();
  }
}
```

### CODE REVIEW

**Comment by aaronjudd** [comment]
@newsiberian I think the changes to productDetails (the react stuff) will be for the 0.12.0 release.  As long it's not too crazy a few updates here and there for 0.11.0 can't hurt ;-)

**Comment by newsiberian** [comment]
@aaronjudd, because of things need to be smooth, I think it was a bad idea to put animation login inside callback. I think I should rollback part of this commit. What do you think?

**Comment by aaronjudd** [comment]
I haven't tested it yet, but looking at the code, yes, I think the animation should be outside of the callback . Also, if there were an error in the callback, then we could then update the animated text with an error message.

File: packages/reaction-core/client/templates/products/productDetail/productDetail.js [hunk 1]
```diff
@@ -161,37 +161,41 @@ Template.productDetail.events({
...
-            if (!error && count === 0) {
-              address = Session.get("address");
-              if (!address) {
-                return locateUser();
+          Meteor.call("cart/addToCart", cartId, productId, currentVariant,
+            quantity, function (error) {
+              let address;
+              if (!error && count === 0) {
+                // todo understand that's going on here
+                //address = Session.get("address");
```
Comment (aaronjudd) [inline packages/reaction-core/client/templates/products/productDetail/productDetail.js]: I think that originally the behaviour was that if you selected an address during an order, that did not save it as your default address in your addressBook - or in the cart.  It was just the selected address for that checkout session.  I think this is probably not needed with the correct behaviour being that the cart specific addressbook selection is now stored in the cart, and the user's default addressbook is in the accounts.

**Comment by aaronjudd** [comment]
re: snippet.  The `locateUser` function in `globals.js`,  is a crude method of inserting default address information into the addAddressBook form.  For example, for me, I get "Santa Monica" "California" defaulted (when it works).  

I'd like to see this swapped out for something cleaner (google address) in the future.
```
/**
 * locateUser
 * @return {Object} set and return session address based on browser latitude, longitude
 */
```

**Comment by aaronjudd** [comment]
@newsiberian I'm closing this PR because the locateUser functionality should remain.  If you want to submit just the productDetail cleanup I'll accept that. If we want to improve the locateUser functionality let's create an issue to discuss some approaches.

**Comment by newsiberian** [comment]
np, but I still suggest you to make this changes manually
here: `packages/reaction-core/client/templates/products/productDetail/productDetail.js`
and here: `packages/reaction-product-simple/client/templates/productDetail/productDetail.js`
```
-      if (currentVariant.parentId === null) {
+      if (typeof currentVariant.parentId !== "string") {
```
this changes is the main reason of this PR. Fix will not allow you to add to cart `product`without selected `option`.

**Comment by aaronjudd** [comment]
@newsiberian resolved in patch.

**Pull request closed without merging**